osm2streets/src/road.rs

```diff
@@ -278,3 +270,48 @@ impl StreetNetwork {
         id
     }
 }
+
+/// The edge of a road, pointed into some intersection
```

Comment:
I just moved this over from the sidewalk corner rendering code. In most of my next-step branches, I want to use this. We should keep adjusting how this works to get closer to the half-road view idea.